openpype/tools/workfiles/app.py

```diff
@@ -37,6 +38,182 @@
+        {
+            "type": "asset",
+            "name": asset_name
+        },
+        {
+            "data.tasks": True,
+            "data.parents": True
+        }
+    )
+
+    task_type = asset_doc["data"]["tasks"].get(task_name, {}).get("type")
+
+    project_task_types = project_doc["config"]["tasks"]
+    task_short = project_task_types.get(task_type, {}).get("short_name")
+
+    asset_parents = asset_doc["data"]["parents"]
+    parent_name = project_doc["name"]
+    if asset_parents:
+        parent_name = asset_parents[-1]
+
+    data = {
+        "project": {
+            "name": project_doc["name"],
+            "code": project_doc["data"].get("code")
+        },
+        "asset": asset_name,
+        "task": {
+            "name": task_name,
+            "type": task_type,
+            "short": task_short,
+        },
+        "parent": parent_name,
+        "version": 1,
+        "user": getpass.getuser(),
+        "comment": "",
+        "ext": None
+    }
+
+    # add system general settings anatomy data
+    system_general_data = get_system_general_anatomy_data()
+    data.update(system_general_data)
+
+    return data
+
+
+class CommentMatcher(object):
+    """Use anatomy and work file data to parse comments from filenames"""
+    def __init__(self, anatomy, template_key, data):
+
+        self.fname_regex = None
+
+        template = anatomy.templates[template_key]["file"]
+        if "{comment}" not in template:
+            # Don't look for comment if template doesn't allow it
+            return
+
+        # Create a regex group for extensions
+        extensions = api.registered_host().file_extensions()
+        any_extension = "(?:{})".format(
+            "|".join(re.escape(ext[1:]) for ext in extensions)
+        )
+
+        # Use placeholders that we can safely escape with regex
+        temp_data = copy.deepcopy(data)
+        temp_data["comment"] = "<<comment>>"
+        temp_data["version"] = "<<version>>"
+        temp_data["ext"] = "<<ext>>"
+
+        formatted = anatomy.format(temp_data)
+        fname_pattern = formatted[template_key]["file"]
+        fname_pattern = re.escape(fname_pattern)
+
+        # Replace comment and version with something we can match with
+        # regex
+        fname_pattern = fname_pattern.replace("<<comment>>", "(.+)")
+        fname_pattern = fname_pattern.replace("<<version>>", "[0-9]+")
+        fname_pattern = fname_pattern.replace("<<ext>>", any_extension)
+
+        # Match from beginning to end of string to be safe
+        fname_pattern = "^{}$".format(fname_pattern)
+
+        self.fname_regex = re.compile(fname_pattern)
+
+    def parse_comment(self, filepath):
+        """Parse the {comment} part from a filename"""
+        if not self.fname_regex:
+            return
+
+        fname = os.path.basename(filepath)
+        match = self.fname_regex.match(fname)
+        if match:
+            return match.group(1)
+
+
+class SubversionLineEdit(QtWidgets.QWidget):
```

Comment:
This is basically similar to the "subset" line edit as in the Creator tool.
Wasn't sure how to implement this simple widget with less code than this.